pretrain/pretrain.py

```diff
@@ -241,12 +254,12 @@ def fit(
             fabric.print(
                 f"iter {metrics['iter']} | step {metrics['step']}: loss {metrics['loss']:.4f}, iter time:"
                 f" {metrics['iter_time'] * 1000:.2f} ms{' (optimizer.step),' if not is_accumulating else ','}"
-                f" remaining time: {metrics['remaining_time'] / 3600 / 24:.2f} days"
+                f" remaining time: {timedelta(seconds=int(metrics['remaining_time']))!s}"
             )
 
             throughput_metrics = throughput.compute()
             metrics.update(throughput_metrics)
-            fabric.log_dict(metrics, step=state["iter_num"])
+            fabric.log_dict(metrics, step=state["iter_num"] - 1)
```

Comment:
But this will mismatch with the other `log_dict` call. Can't we log both on the same iteration?